extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -423,30 +423,18 @@ static public function setmainDomain ( $city_id, $domain = null, $reason = null
 	 * @access public
 	 * @static
 	 *
-	 * @param $url string - domain name
+	 * @param $url string - domain name in form:
+	 * http://community-name.wikia.com
 	 *
 	 * @return integer - id of domain or null if not found
 	 */
 	static public function UrlToID( $url ) {
-
 		$city_id = false;
 		$parts = parse_url( $url );
 		if ( isset( $parts[ "host" ] ) ) {
 			$host = self::getDomainHash( $parts[ "host" ] );
-
-			// TODO: Eloy: Can this hack be removed?
-			if ( $host === "memory-alpha.org" ) {
-				/**
-				 * for memory-alpha check first element of path
-				 */
-				$parts = explode( "/", $parts[ "path" ] );
-				$host = sprintf( "%s.%s", $parts[ 1 ], $host );
-				$city_id = self::DomainToId( $host );
-			}
-			else {
-				$host = preg_replace('/^(?:preview\.|verify\.)/i', '', $host);
-				$city_id = self::DomainToId( $host );
-			}
+			$host = preg_replace('/^(?:preview\.|verify\.)/i', '', $host);
+			$city_id = self::DomainToId( $host );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)